test/test_app.py

```diff
@@ -1,29 +1,66 @@
-import os
-import tempfile
+import json
+from test.auth_for_testing import get_user_token_header
+from test.database_for_testing import with_test_db
 
 import pytest
 
 from boxwise_flask.app import db
+from boxwise_flask.models import Camps, Cms_Usergroups_Camps, Cms_Users, Person
 from boxwise_flask.routes import app
 
+MODELS = (Person, Camps, Cms_Usergroups_Camps, Cms_Users)
+
 
 @pytest.fixture
 def client():
-    db_fd, db_filepath = tempfile.mkstemp(suffix=".sqlite3")
-    app.config["DATABASE"] = {
-        "name": db_filepath,
-        "engine": "peewee.SqliteDatabase",
-    }
-    db.init_app(app)
+    """test client to setup app"""
+    app.config["DATABASE"] = "sqlite:///:memory:"
 
     with app.test_client() as client:
         yield client
 
-    os.close(db_fd)
-    os.remove(db_filepath)
+    db.init_app(app)
```

Comment:
Is it intended that this is called after the block before? I thought that the `yield` statement would block execution, meaning that `db.init_app()` is never called when the client fixture is used.

Reply:
I was very confused about this as well. it seems to work either side of the block.